src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -87,14 +91,48 @@ private void addSearchPathsToClassLoader(FunctionMethodDescriptor function) thro
 		function.getLibDirectory().ifPresent(d -> registerWithClassLoaderProvider(d));
 	}
 
-	private void registerWithClassLoaderProvider(File libDirectory) {
+	void registerWithClassLoaderProvider(File libDirectory) {
 		try {
-			classLoaderProvider.addDirectory(libDirectory);
+			if(SystemUtils.IS_JAVA_1_8) {
+				String workerLibPath = System.getenv(Constants.FUNCTIONS_WORKER_DIRECTORY) + "/lib";
+				String javaReverseLibLoading = System.getenv(Constants.JAVA_REVERSE_LIB_LOADING);
+				File workerLib = new File(workerLibPath);
+				if(javaReverseLibLoading != null && javaReverseLibLoading.toLowerCase().equals("true")) {
+					if(checkLibFolder(workerLib, workerLibPath, "CLIENT_LIB_FIRST")) {
+						classLoaderProvider.addDirectory(libDirectory);
+						classLoaderProvider.addDirectory(workerLib);
+					}
+				} else {
+					if(checkLibFolder(workerLib, workerLibPath, "WORKER_LIB_FIRST")) {
+						classLoaderProvider.addDirectory(workerLib);
+						classLoaderProvider.addDirectory(libDirectory);
+					}
+				}
+			} else {
+				classLoaderProvider.addDirectory(libDirectory);
+			}
 		} catch (Exception ex) {
 			ExceptionUtils.rethrow(ex);
 		}
 	}
 
+	boolean checkLibFolder(File workerLib, String workerLibPath, String status) throws Exception{
+		if(!workerLib.exists()) {
+			throw new Exception("Error loading worker jars, for "+ status +", from path:  " + workerLibPath);
```

Comment:
Is this always guaranteed to exist? Also, instead of generic exception,  throw FileNotFoundException?

Reply:
Will change it to FileNotFoundException. 
You bring good point, What is the other cases that can cause it not to exit?